content/guest-post/nestjs-authentication-with-loginradius-api/index.md

```diff
@@ -0,0 +1,452 @@
+API_SECRET=,your api secret>
+SOTT= <your sott>
+```
+
+## Authentication Flow
+
+In this project, you will be authenticating the user using their email and password. The following are the major steps you will be following:
+
+- Create a signup route that will get the user information and create a new user.
+- Create a login route that accepts the email and password of the user. This data will be authenticated, and an access token will be sent back in the response.
+- Create a protected route that will only accept requests with valid access tokens.
+
+## Create an Authentication Module
+
+Generate an auth module, controller, and service by running the following code.
+
+```bash
+nest generate module auth
+nest generate controller auth
+```
+
+## Create a User
+
+To create a user, you need to create a signup route that will accept the email and password.
+
+Since you are using TypeScript, define the DTO (Data Transfer Object) schema to validate the user data passed in the request body.
+
+In the `auth` folder, add the `dto` folder and create a `UserDTO` class in the `user.dto.ts` file.
+
+```JavaScript
+export class UserDto {
+  email: string,
+  password: string
+}
+```
+
+Next, inside the `AuthController`, import the DTO to be used to validate the request body.
+
+```JavaScript
+import { Injectable } from '@nestjs/common';
+import { UserDto } from './dto/user.dto';
+
+@Injectable()
+@Controller('auth')
+export class AuthController {
+  @Post('signup')
+  async signup(@Body() registerUserDto: UserDto) {
+    // Register user
+  }
+}
+```
+
+### Generate Auth Service File
+
+A service file is used to abstract the business logic away from the controller. You will be handling the actual authentication and authorization process in this file.
+
+Generate a service for auth by running the following command.
+
+```bash
+nest g service auth
+```
+
+Next, populate the auth.service file by adding the signup method.
+
+```JavaScript
+import { Injectable } from '@nestjs/common';
+import { UserDto } from './dto/user.dto';
+import * as LRAuthPrrovider from 'loginradius-sdk'
+
+@Injectable()
+export class AuthService {
+    async signup(registerUserDto: UserDto) {
+      console.log('sign up')
+    }
+}
+```
+
+Note that you are also importing the user DTO and the loginradius-sdk at the top of the file.
+To execute the signup method in the signup route, inject it in the `auth.contoller.ts` file.
+
+```JavaScript
+import { Injectable } from '@nestjs/common';
+import { UserDto } from './dto/user.dto';
+import { AuthService } from './auth.service';
+
+@Injectable()
+@Controller('auth')
+export class AuthController {
+  constructor(private readonly authService: AuthService) { }
+
+  @Post('signup')
+  async signup(@Body() registerUserDto: UserDto) {
+    let response = await this.authService.signup(registerUserDto)
+    return response
+  }
+}
+```
+
+Before registering the user, validate if the email is already in use.
+
+Start by configuring loginradius-sdk.
+
```

Comment:
Before importing, add a line stating the command used to install `loginradius-sdk`, i.e `npm i loginradius-sdk`